Description: Luis Uribe fixed autolog.init file in order to be LSB-compliant. 
Marcos Fouces added support for "status" option, modernized functions and fixed (de)installation issues on Debian systems.
Author: Luis Uribe <acme@eviled.org>, Marcos Fouces <marcos.fouces@gmail.com>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=850761
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=605890
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/autolog.init
+++ b/autolog.init
@@ -1,44 +1,106 @@
 #! /bin/sh
-# Copyright (c) 2000 Carsten Juerges, Hannover, Germany.  All rights reserved.
 #
-# Author: Carsten Juerges <juerges@cip-bau.uni-hannover.de>, 2000
+# autolog: Terminates connections for idle users
 #
-# /sbin/init.d/autolog
+### BEGIN INIT INFO
+# Provides:          autolog
+# Required-Start:    $remote_fs
+# Required-Stop:     $remote_fs
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Start the autolog daemon firewall at boot time
+# Description:       Script to start/stop/reload the autolog daemon
+### END INIT INFO
 #
+# Copyright (c) 2000 Carsten Juerges, Hannover, Germany.  All rights reserved.
+# Author: Carsten Juerges <juerges@cip-bau.uni-hannover.de>, 2000
+# Debian-ized by Paul Telford <paul@droflet.net> 01/30/2003
+# Make LSB-compliant on 01/06/2007 by Luis Uribe <acme@eviled.org> 
+# Modernize functions and fix Debian (de)install issues on 2017 by Marcos Fouces <mfouces@yahoo.es>
+
+PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
+DAEMON=/usr/sbin/autolog
+NAME=autolog
+DESC="terminates connections for idle users"
+PIDFILE=/var/run/$NAME.pid
+SCRIPTNAME=/etc/init.d/$NAME
+
+. /lib/init/vars.sh
+
+. /lib/lsb/init-functions
+
+test -x /usr/sbin/autolog || exit 0
+
+do_start()
+{
+	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON --test > /dev/null \
+		|| return 1
+	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON -- \
+		$DAEMON_ARGS \
+		|| return 2
+}
+
+do_stop()
+{
+	start-stop-daemon --stop --quiet --retry=TERM/30/KILL/5 --pidfile $PIDFILE --name $NAME
+	RETVAL="$?"
+	[ "$RETVAL" = 2 ] && return 2
+
+	start-stop-daemon --stop --quiet --oknodo --retry=0/30/KILL/5 --exec $DAEMON
+	[ "$?" = 2 ] && return 2
+	rm -f $PIDFILE
+	return "$RETVAL"
+}
+
+do_reload() {
+
+	start-stop-daemon --stop --signal 1 --quiet --pidfile $PIDFILE --name $NAME
+	return 0
+}
 
-. /etc/rc.config
-
-base=${0##*/}
-link=${base#*[SK][0-9][0-9]}
-
-test $link = $base && START_AUTOLOG=yes
-test "$START_AUTOLOG" = yes || exit 0
-
-export PATH=/bin:/usr/bin:/sbin:/usr/sbin
-export HOME=/
-
-return=$rc_done
 case "$1" in
-    start)
-	echo -n "Starting autolog daemon"
-	startproc /usr/sbin/autolog || return=$rc_failed
-	echo -e "$return"
+  start)
+	[ "$VERBOSE" != no ] && log_daemon_msg "Starting $DESC" "$NAME"
+	do_start
+	case "$?" in
+		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
+		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
+	esac
+	;;
+  stop)
+	[ "$VERBOSE" != no ] && log_daemon_msg "Stopping $DESC" "$NAME"
+	do_stop
+	case "$?" in
+		0|1) [ "$VERBOSE" != no ] && log_end_msg 0 ;;
+		2) [ "$VERBOSE" != no ] && log_end_msg 1 ;;
+	esac
 	;;
-    stop)
-	echo -n "Shutting down autolog daemon"
-	killproc -TERM /usr/sbin/autolog || return=$rc_failed
-	echo -e "$return"
+  status)
+	status_of_proc "$DAEMON" "$NAME" && exit 0 || exit $?
 	;;
-    restart)
-	$0 stop && $0 start || return=$rc_failed
+
+  restart|force-reload)
+
+	log_daemon_msg "Restarting $DESC" "$NAME"
+	do_stop
+	case "$?" in
+	  0|1)
+		do_start
+		case "$?" in
+			0) log_end_msg 0 ;;
+			1) log_end_msg 1 ;; 
+			*) log_end_msg 1 ;;
+		esac
+		;;
+	  *)
+		log_end_msg 1
+		;;
+	esac
 	;;
-    status)
-	checkproc /usr/sbin/autolog && echo "autolog up" || echo "No autolog"
+  *)
+	echo "Usage: $SCRIPTNAME {start|stop|status|restart|force-reload}" >&2
+	exit 3
 	;;
-    *)
-	echo "Usage: $0 {start|stop|status|restart}"
-	exit 1
 esac
 
-test "$return" = "$rc_done" || exit 1
-exit 0
+:
